Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +15,214 @@ public static class BalanceStats
+        {
+            if (!IsOptionValid(option)) throw new Exception($"{option} is not a valid option for function ProcessPeopleBalance().");
+
+            var (name, money) = ProcessPeopleBalance(peopleAndBalances, option);
+            var people = new List<string>();
+            var moneyToCompare = (option == "max" || option == "rich") ? money.Max() : money.Min();
+
+            for (var i = 0; i < money.Count; i++)
+            {
+                if (money[i] == moneyToCompare) people.Add(name[i]);
+            }
+
+            var (processedNames, isPlural) = ProcessName(people);
+
+            return (processedNames, moneyToCompare, isPlural);
+        }
+
+        private static (List<string> name, List<decimal>) ProcessPeopleBalance(string[] peopleAndBalances, string option)
+        {
+            var nameList = new List<string>();
+            var moneyProcessed = new List<decimal>();
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                try
+                {
+                    var moneyList = CheckData(peopleAndBalances[i], out var name);
+                    if (string.IsNullOrEmpty(name)) throw new Exception(Constants.GenericMessageError);
+
+                    switch (option)
+                    {
+                        case "max" when moneyList.Count == 0:
+                            continue;
+                        case "max":
+                            moneyProcessed.Add(moneyList.Max());
+                            break;
+                        case "min" when moneyList.Count == 0:
+                            continue;
+                        case "min":
+                            moneyProcessed.Add(CheckBalance(moneyList, option));
+                            break;
+                        case "rich":
+                        case "poor":
+                        {
+                            if (moneyList.Count < 1) continue;
+                            moneyProcessed.Add(moneyList[^1]);
+                            break;
+                        }
+                    }
+
+                    nameList.Add(name);
+                }
+                catch (Exception ex)
+                {
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine(ex);
+                    Console.WriteLine($"The following data from index: {i} is not valid, skipping.");
+                    Console.WriteLine($"\"{peopleAndBalances[i]}\"");
+                    Console.ResetColor();
+                    continue;
+                }
+            }
+
+            if (nameList.Count != moneyProcessed.Count || nameList == null || moneyProcessed == null || nameList.Count == 0 || moneyProcessed.Count == 0)
+            {
+                throw new Exception($"Something went very wrong in function ProcessPeopleBalance().");
```

Comment:
If you do throw an exception, don't throw default exception. Make your own, custom exception. Also, something went wrong is not very helpful, in fact it's worse that not handling the exception (because you don't leave any traces for the error that happened. Normally the error should be logged and rethrown or extra details provided, but never less details provided. This might have been an imitation for giving a user friendly error, but you still throw an exception, so that is not the case.